[bugfix] Filter model metadata by current widget selection

christian-byrne opened the pull request:
Filter models in workflow metadata to only include those actually selected in node widgets. Prevents false positives in the missing models dialog when nodes contain outdated metadata.

Fixes #3664

┆Issue is synchronized with this [Notion page](https://www.notion.so/PR-4021-bugfix-Filter-model-metadata-by-current-widget-selection-2036d73d365081bdb416c3c3f584d4a4) by [Unito](https://www.unito.io)

christian-byrne committed:
[bugfix] Filter model metadata by current widget selection

Only include model metadata for models actually selected in node widgets. This prevents false positives in the missing models dialog when nodes contain outdated metadata from previously selected models.

Fixes #3664

christian-byrne committed:
improve comments clarity

webfiltered reviewed (commented):
This is really just one suggested change.

"Simplify code using idiomatic undefined"

christian-byrne committed:
Update src/scripts/app.ts

Co-authored-by: filtered <176114999+webfiltered@users.noreply.github.com>

christian-byrne committed:
Update src/utils/modelMetadataUtil.ts

Co-authored-by: filtered <176114999+webfiltered@users.noreply.github.com>

christian-byrne committed:
Update src/utils/modelMetadataUtil.ts

Co-authored-by: filtered <176114999+webfiltered@users.noreply.github.com>

christian-byrne committed:
Update src/utils/modelMetadataUtil.ts

Co-authored-by: filtered <176114999+webfiltered@users.noreply.github.com>

christian-byrne commented on src/utils/modelMetadataUtil.ts:
```diff
@@ -0,0 +1,52 @@
+  type: string
+  widgets_values?: unknown[] | Record<string, unknown>
+  properties?: { models?: ModelFile[] }
+}): ModelFile[] {
```
Thanks!

christian-byrne committed:
Update src/utils/modelMetadataUtil.ts

Co-authored-by: filtered <176114999+webfiltered@users.noreply.github.com>

christian-byrne committed:
[bugfix] Fix TypeScript errors in modelMetadataUtil tests

Add non-null assertions for test cases expecting arrays and correct test expectations for cases that should return undefined.

christian-byrne commented:
Changes:

- Merged changes requested in above review comments
- Updated unit test expectations to expect `undefined` rather than empty array to match above changes

webfiltered reviewed (approved):
Will merge into 1.22
